app/controllers/declaration/AdditionalFiscalReferencesController.scala

```diff
@@ -18,99 +18,62 @@ package controllers.declaration
+    anotherYesNoForm
+      .bindFromRequest()
       .fold(
-        formWithErrors => Future.successful(badRequest(mode, itemId, formWithErrors, cachedData.references)),
-        updatedCache =>
-          updateExportsCache(itemId, AdditionalFiscalReferencesData(updatedCache))
-            .map(_ => navigator.continueTo(mode, routes.AdditionalFiscalReferencesController.displayPage(_, itemId)))
+        (formWithErrors: Form[YesNoAnswer]) =>
+          BadRequest(
+            additionalFiscalReferencesPage(
+              mode,
+              itemId,
+              formWithErrors,
+              cachedAdditionalReferencesData(itemId).map(_.references).getOrElse(Seq.empty)
+            )
+        ),
+        validYesNo =>
+          validYesNo.answer match {
+            case YesNoAnswers.yes =>
+              navigator.continueTo(mode, controllers.declaration.routes.AdditionalFiscalReferencesAddController.displayPage(_, itemId))
+            case YesNoAnswers.no => navigator.continueTo(mode, routes.CommodityDetailsController.displayPage(_, itemId))
+        }
       )
-
-  private def saveAndContinue(mode: Mode, itemId: String, form: Form[AdditionalFiscalReference], cachedData: AdditionalFiscalReferencesData)(
-    implicit request: JourneyRequest[AnyContent]
-  ): Future[Result] =
-    MultipleItemsHelper
-      .saveAndContinue(form, cachedData.references, isMandatory = true, limit)
-      .fold(
-        formWithErrors => Future.successful(badRequest(mode, itemId, formWithErrors, cachedData.references)),
-        updatedCache =>
-          if (updatedCache != cachedData.references)
-            updateExportsCache(itemId, AdditionalFiscalReferencesData(updatedCache))
-              .map(_ => navigator.continueTo(mode, routes.CommodityDetailsController.displayPage(_, itemId)))
-          else Future.successful(navigator.continueTo(mode, routes.CommodityDetailsController.displayPage(_, itemId)))
-      )
-
-  def removeReference(mode: Mode, itemId: String, value: String) = itemAction(itemId).async { implicit request =>
-    val cacheModel = request.item.additionalFiscalReferencesData
-      .getOrElse(AdditionalFiscalReferencesData(Seq.empty))
-    val updatedCache = cacheModel.removeReference(value)
-    updateExportsCache(itemId, updatedCache).map {
-      case Some(_) =>
-        navigator.continueTo(mode, routes.AdditionalFiscalReferencesController.displayPage(_, itemId))
-      case None => navigator.continueTo(mode, routes.ItemsSummaryController.displayItemsSummaryPage(_))
-    }
   }
 
-  private def badRequest(mode: Mode, itemId: String, formWithErrors: Form[AdditionalFiscalReference], references: Seq[AdditionalFiscalReference])(
-    implicit request: JourneyRequest[AnyContent]
-  ): Result =
-    BadRequest(additionalFiscalReferencesPage(mode, itemId, formWithErrors, references))
+  private def anotherYesNoForm: Form[YesNoAnswer] = YesNoAnswer.form(errorKey = "declaration.additionalFiscalReferences.add.another.empty")
```

Comment:
Nit: The name was misleading to me. When I read it first time I though "just another Yes/No form". Maybe `addAnotherYesNoForm` would be more clear?